exp/addons/internal/controllers/clusterresourceset_controller_test.go

```diff
@@ -791,4 +791,219 @@ data:
 			return resource.Hash != oldHash
 		}, timeout).Should(BeTrue())
 	})
+
+	t.Run("Should create ClusterResourceSet with strategy 'AlwaysApply' and reconcile configmap only once if data has not changed", func(t *testing.T) {
```

Comment:
Did this test failed before making the changes in exp/addons/internal/controllers/clusterresourceset_controller.go?

Reply:
This is a work in progress. I have been trying to figure out how to tell when things trigger and how things are changing when. I am getting close, with the guidance of @dlipovetsky.

Reply:
Yep will open another branch where it fails so you can verify it.